store/src/java/com/zimbra/cs/filter/jsieve/EditHeaderExtension.java

```diff
@@ -564,4 +566,35 @@ private boolean matchValue(String regex, String value) {
         Matcher matcher = pattern.matcher(value);
         return matcher.matches();
     }
+
+    static public boolean isTolerableFormedMessage(ZimbraMailAdapter zma, String actionName, MimeMessage mm) {
```

Comment:
if saveChanges fails due to some error other than ParseException, the mime gets updated or not? if not, then calling saveChanges once after making changes should be sufficient.